openstack_exporter/collectors/cinderbackend.py

```diff
@@ -21,18 +21,43 @@
 
 LOG = logging.getLogger('openstack_exporter.exporter')
 
+DEFAULT_VALID_SHARDING_BACKENDS = ['standard_hdd', 'vmware']
+DEFAULT_VALID_NO_SHARD_BACKENDS = []
+ALLOW_UNEXPECTED_BACKENDS_DEFAULT = False
+
 
 class CinderBackendCollector(BaseCollector.BaseCollector):
-    version = "1.0.2"
+    version = "1.0.3"
 
-    def __init__(self, openstack_config):
-        super().__init__(openstack_config)
+    def __init__(self, openstack_config, collector_config):
+        super().__init__(openstack_config, collector_config)
         # We have to create a cinder client here
         # because the openstacksdk doesn't currently
         # Support the quota functions.
         self.cinder_client = self._cinder_client()
         self.labels = ['backend', 'pool', 'shard']
         LOG.debug(f"Openstack Exporter CinderBackend Version {self.version}")
+        LOG.debug(f"Collector configuration {self.collector_config}")
+        # Make sure there is a default setting for valid_backends
+        cinder_config = self.collector_config['cinderbackend']
+        if 'expected_sharding_backends' not in cinder_config:
+            self.expected_sharding_backends = DEFAULT_VALID_SHARDING_BACKENDS
+        else:
+            backends = cinder_config['expected_sharding_backends']
+            self.expected_sharding_backends = backends.split(',')
```

Comment:
I think this needs a `strip()`, because the example confing contains `, ` as separator, i.e. `[b.strip() for b in backends.split(',')]`

Reply:
done